IAmLegend/Assets/Scripts/GameManager.cs

```diff
@@ -52,8 +66,68 @@ void SpawnZombies()
         playerLastPosition = playerCurPosition;
     }
 
-    public void addScore(int points)
+    public void AddScore(int points)
     {
         score += points;
     }
+
+    private void EndGame()
+    {
+        HealthSystem playerHealth = player.GetComponents<HealthSystem>()[0];
+        HealthSystem dogHealth = dog.GetComponents<HealthSystem>()[0];
+        // If the player is dead (lose) 
+        if(playerHealth.GetHealth() <= 0)
+        {
+            // If the dog is alive, triple score
+            if(dogHealth.GetHealth() > 0)
+            {
+                score *= 3;
+            }
+            scoreDeadScreenTMP.SetText(score.ToString());
+            deadScreen.SetActive(true);
+            Time.timeScale = 0;
+        }
+        // If it's 6:00 AM (won)
+        if(timer >= 360)
+        {
+            // If the dog is alive, triple score
+            if(dogHealth.GetHealth() > 0)
+            {
+                score *= 3;
+            }
+            scoreWinScreenTMP.SetText(score.ToString());
+            winScreen.SetActive(true);
+            Time.timeScale = 0;
+        }
+    }
+
+    private void Sunrise()
+    {
+        // Sunrise from 5:20 AM to 6:00 AM
+        Light lightSource = directionalLight.GetComponents<Light>()[0];
+        
+        int timeMinutes = Mathf.FloorToInt(timer / 60F);
+        int timeSeconds = Mathf.FloorToInt(timer - timeMinutes * 60);
+        
+        // If 5:30AM, increase sunlight and lerp to a white light source
+        if(lightSource.intensity < 1  && timeMinutes >= 5 &&  timeSeconds > 20)
+        {
+            lightSource.intensity += 0.001f;
+            lightSource.color =  Color.Lerp(lightSource.color, Color.white, Mathf.PingPong(Time.time, 1));

```

Comment:
It might be a good idea to dim the torch. Its too much light and the zombies look weird. I have created a seperate issue to do it if we have time #70 . You don't have to do it here